### PR TITLE
Fix before/afterHook parameters for Cucumber

### LINKED ISSUE #7454: cucumber js world context is undefined in hooks.ts
# Issue description
cucumber js world context is undefined in hooks.ts

# Steps to reproduce
`beforeHook: function (test, context, stepData, world) {console.log(world)},` in hooks.ts

# Expected behavior
I expect I'd log the world context of my cucumber tests

# Actual behavior
the output is undefined.

# Issue occurred on browser/platform 
selenium, wdio, cucumber framework. Didn't modify anything else about the project.

### PR DESCRIPTION
**Pull request by christian-bromann**
## Proposed changes

fixes #7454

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/wdio-jasmine-framework/src/index.ts [hunk 1]
```diff
@@ -351,7 +351,7 @@ class JasmineAdapter {
         return function (this: jasmine.Spec, passed: boolean, data: ResultHandlerPayload) {
             try {
                 expectationResultHandler!.call(this, passed, data)
-            } catch (e) {
+            } catch (e: any) {
```
Comment (erwinheitzman) [inline packages/wdio-jasmine-framework/src/index.ts:354]:
```suggestion
            } catch (e: Error) {
```

File: packages/wdio-utils/src/test-framework/testFnWrapper.ts [hunk 2]
```diff
@@ -76,7 +76,7 @@ export const testFrameworkFnWrapper = async function (
     const testStart = Date.now()
     try {
         result = await promise
-    } catch (err) {
+    } catch (err: any) {
```
Comment (erwinheitzman) [inline packages/wdio-utils/src/test-framework/testFnWrapper.ts:79]:
```suggestion
    } catch (err: Error) {
```

File: packages/wdio-types/src/Services.ts [hunk 3]
```diff
@@ -153,11 +153,9 @@ export interface HookFunctions {
      * Hook that gets executed _before_ a hook within the suite starts (e.g. runs before calling
      * beforeEach in Mocha). `stepData` and `world` are Cucumber framework specific properties.
      * @param test      details to current running test (or step in Cucumber)
-     * @param context   context to current running test
-     * @param stepData  Cucumber step data
-     * @param world     Cucumber world
+     * @param context   context to current running test (or World object in Cucumber)
```
Comment (erwinheitzman) [inline packages/wdio-types/src/Services.ts:156]:
```suggestion
     * @param context   context to current running test (represents World object in Cucumber)
```

**Review by erwinheitzman — COMMENTED** [review]
Left some feedback 👍

File: packages/wdio-jasmine-framework/src/index.ts (hunk 1 above)
Comment (christian-bromann) [reply packages/wdio-jasmine-framework/src/index.ts:354]: Catch clause variable type annotation must be 'any' or 'unknown' if specified.

File: packages/wdio-utils/src/test-framework/testFnWrapper.ts (hunk 2 above)
Comment (christian-bromann) [reply packages/wdio-utils/src/test-framework/testFnWrapper.ts:79]: Catch clause variable type annotation must be 'any' or 'unknown' if specified.

**Pull request merged**